Sources/RandomAccessCollection+Sweet.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+extension CollectionType {
+    func enumerateWithNext() -> [(Any, Any?)] {
```

Comment:
If you use Any that means that when you call this function you first need to cast it to the type you need right? Can't we make it so that it infers the type from the type of the collection? With <T> something? It's vaguely in my memory, but hard to find on internet!